### PR TITLE
let update_synapse_database run on a multi-database configurations

### PR DESCRIPTION
**Pull request by thefinn93**
The `update_synapse_database` script checks for the presence of the `database` key in the config. When multiple databases are used, the rest of synapse expects a different key, `databases`, which is mutually exclusive with `database`. this fixes the check to allow either key.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: changelog.d/13422.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+let update_synapse_database run in configurations with multiple databases. Contributed by @thefinn93 @ Beeper.
```
Comment (clokep) [inline changelog.d/13422.bugfix:1]:
```suggestion
Fix a long-standing bug where the `update_synapse_database` script could not be run with multiple databases. Contributed by @thefinn93 @ Beeper.
```

File: synapse/_scripts/update_synapse_database.py [hunk 2]
```diff
@@ -97,8 +97,8 @@ def main() -> None:
     # Load, process and sanity-check the config.
     hs_config = yaml.safe_load(args.database_config)
 
-    if "database" not in hs_config:
-        sys.stderr.write("The configuration file must have a 'database' section.\n")
+    if "database" not in hs_config and "databases" not in hs_config:
+        sys.stderr.write("The configuration file must have a 'database' or 'databases' section.\n")
         sys.exit(4)
```
Comment (clokep) [inline synapse/_scripts/update_synapse_database.py:102]: I think we do similar (and more thorough) error checking when we parse the config anyway... I wonder if we can just get rid of these lines of code?

https://github.com/matrix-org/synapse/blob/493c2fc44abcf3457953cc2f6f23509ff7855253/synapse/config/database.py#L97-L123

Comment (thefinn93) [reply synapse/_scripts/update_synapse_database.py:102]: I wasn't clear what this check was for. it does seem like there is another check lower down. Should I remove it?

Comment (thefinn93) [reply synapse/_scripts/update_synapse_database.py]: I went ahead and removed it

Comment (clokep) [reply synapse/_scripts/update_synapse_database.py]: Yeah, I think it is fine to remove it. The error message seems pretty much the same too.

**Comment by clokep** [comment]
Putting this back in the queue since I've lost all state on it and likely won't get to it soon.

File: synapse/_scripts/update_synapse_database.py [hunk 3]
```diff
@@ -97,10 +96,6 @@ def main() -> None:
     # Load, process and sanity-check the config.
     hs_config = yaml.safe_load(args.database_config)
 
-    if "database" not in hs_config:
-        sys.stderr.write("The configuration file must have a 'database' section.\n")
-        sys.exit(4)
-
```
Comment (squahtx) [inline synapse/_scripts/update_synapse_database.py:103]: Does this mean we no longer error when both `database` and `databases` are missing?

Also does anything rely on the exit code being 4?

Comment (clokep) [reply synapse/_scripts/update_synapse_database.py:103]: I think we do this checking at:

https://github.com/matrix-org/synapse/blob/493c2fc44abcf3457953cc2f6f23509ff7855253/synapse/config/database.py#L97-L123

Maybe I missed something though!

**Review by DMRobertson — APPROVED** [review]
I think
- this seems sane and worthwhile
- I think it's unlikely anything is relying on error code 4
- Patrick's comments about the config parser enforcing that we have `databases` xor `database` fields seem sane to me too.

I'd like to merge it.

@thefinn93 were you able to test this change?  I'd just like some reassurance that this works. (I can't see why it wouldn't---we call hs.setup() just like we would at startup).

**Comment by thefinn93** [comment]
yes, this change was tested and didn't seem to cause any issues

**Pull request merged**